SConstruct

```diff
@@ -215,52 +266,138 @@ Default(environ.get('M5_DEFAULT_BINARY', 'build/ARM/gem5.debug'))
 #
 ########################################################################
 
-# helper function: find last occurrence of element in list
-def rfind(l, elt, offs = -1):
-    for i in range(len(l)+offs, 0, -1):
-        if l[i] == elt:
-            return i
-    raise ValueError("element not found")
+kconfig_actions = (
+    'defconfig',
+    'guiconfig',
+    'listnewconfig',
+    'menuconfig',
+    'oldconfig',
+    'olddefconfig',
+    'savedefconfig',
+    'setconfig',
+)
+
+Help("""
+Kconfig:
+        In addition to the default configs, you can also create your own
+        configs, or edit one that already exists. To use one of the kconfig
+        tools with a particular directory, use a target which is the directory
+        to configure, and then the name of the tool. For example, to run
+        menuconfig on directory build/foo/bar, run:
+
+        scons menuconfig build/foo/bar
+
+        will set up a build directory in build/foo/bar if one doesn't already
+        exist, and open the menuconfig editor to view/set configuration
+        values.
+
+        The tools available for working with kconfig are generally very
+        similar to ones used with the linux kernel, so information about the
+        kernel versions will typically (but not always) apply here as well.
```

Comment:
> will typically (but not always) apply here

When does kernel documentation *not* apply here? That's going to be important to document.

Reply:
Deleted